owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/SaveFileUseCase.kt

```diff
@@ -0,0 +1,32 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Christian Schabesberger
+ * Copyright (C) 2020 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.domain.files.usecases
+
+import com.owncloud.android.domain.BaseUseCaseWithResult
+import com.owncloud.android.domain.files.FileRepository
+import com.owncloud.android.domain.files.model.OCFile
+
+class SaveFileOrFolderUseCase(
+    private val fileRepository: FileRepository
+) : BaseUseCaseWithResult<Unit, SaveFileOrFolderUseCase.Params>() {
+    override fun run(params: Params): Unit = fileRepository.saveFile(params.fileToUpload)
+
+    data class Params(val fileToUpload: OCFile)
```

Comment:
We will use this usecase when we download a file for example and we want to save its properties into the database. So, from my point of view, fileToUpload is not the most accurated name, what do you think?